scripts/legislation/africa/uganda_english.sh

```diff
@@ -0,0 +1,287 @@
+#!/bin/bash
+
+function remove_all_text_before_first_header {
+  sed -E 's/PART I - PRELIMINARY/\nPART I - PRELIMINARY/' | \
+    sed -n '/PART I - PRELIMINARY/,$p'
+}
+
+function prefix_article_numbers_with_article_literal {
+  sed -E 's/^([0-9]+\.)/Article \1/'
+}
+
+function replace_parentheses_around_article_delimiters_with_square_brackets {
+  sed -E 's/^\(([A-Za-z0-9]+)\)/[\1]/' 
+}
+
+function remove_all_text_after_last_article {
+  sed -E '/^\(109\)/q'
+}
+
+function amend_errors_in_headers {
+  sed -E 's/PART/\n\nPART/g' | \
+    sed -E 's/^(PART .*)\[•\]/\1-/' | \
+    sed -n '/PART I /,$p' | \
+    sed -E 's/ARRANGEMENTS The Authority/ARRANGEMENTS\n\nThe Authority/' 
+}
+
+function amend_errors_in_articles {
+  sed -E 's/ \[•\]/:/g' | \
+    sed -E 's/([a-z])- /\1: /g' | \
+    sed -E 's/\(([a-z])\)/[\1]/g' | \
+    sed -E 's/ \(([0-9])\)/ [\1]/g' | \
+    sed -E 's/([0-9]+)\(([0-9]+)\)/\1[\2]/g' | \
```

Comment:
They are necessary because `replace_parentheses_around_article_delimiters_with_square_brackets` needs to be invoked before `apply_common_transformations_to_stdin`. However, after `apply_common_transformations_to_stdin` is invoked there are still cases of brackets around numbers and letters that should be square brackets. Lines 30-32 remedy these cases.

However, line 32 is not necessary for this legislation because there is no formatting in this legislation that matches the intended use of this function. I.e. changing "paragraph 32(3)" to paragraph 32[3]".

Reply:
The lingering parentheses around numbers and letters might be attributable to the fact that the function on line 13 will only replace the first match on a given line. If you want the function to replace all matches on a given line, then you need to add the 'g' flag, i.e.

```
sed -E 's/^\(([A-Za-z0-9]+)\)/[\1]/g'
```